fs/rc/rcserver/rcserver_test.go

```diff
@@ -539,6 +539,151 @@ func makeMetricsTestCases(stats *accounting.StatsInfo) (tests []testRun) {
 	return
 }
 
+func TestShare(t *testing.T) {
+	tests := []testRun{
+		{
+			Name:   "share-object-randomToken-unlimited",
+			URL:    "share/create",
+			Method: "POST",
+			Body: `{
+				"fs": "testdata/files",
+				"remote": "file.txt"
+			}`,
+			ContentType: "application/json",
+			Status:      http.StatusOK,
+			Contains: regexp.MustCompile(`{
+	"sharedLink": "share/links/.*/file.txt"
+}`),
+		}, {
+			Name:        "share-list",
+			URL:         "share/list",
+			Method:      "POST",
+			Body:        `{}`,
+			ContentType: "application/json",
+			Status:      http.StatusOK,
+			Contains: regexp.MustCompile(`{
+	"sharedLinks": \[
+		{
+			"fs": "testdata/files",
+			"remote": "file.txt",
+			"sharedName": "file.txt",
+			"token": ".*"
+		}
+	\]
+}
+`),
+		}, {
+			Name:   "share-object-not-found",
+			URL:    "share/create",
+			Method: "POST",
+			Body: `{
+				"fs": "testdata/files",
+				"remote": "file2.txt"
+			}`,
+			ContentType: "application/json",
+			Status:      http.StatusNotFound,
+			Expected: `{
+	"error": "object not found",
+	"input": {
+		"fs": "testdata/files",
+		"remote": "file2.txt"
+	},
+	"path": "share/create",
+	"status": 404
+}
+`,
+		}, {
+			Name:   "share-specify-token",
+			URL:    "share/create",
+			Method: "POST",
+			Body: `{
+				"fs": "testdata/files",
+				"remote": "file.txt",
+				"token": "123456"
+			}`,
+			ContentType: "application/json",
+			Status:      http.StatusOK,
+			Expected: `{
+	"sharedLink": "share/links/123456/file.txt"
+}
+`,
+		}, {
+			Name:     "share-get-content",
+			URL:      "share/links/123456/file.txt", // TODO: it can be fetched without Authentication Header, btw can't test that here
```

Comment:
I don't know how to set up the test for fetching data without  Authentication Header

Reply:
@ncw @negative0  Are there examples in the integration suite?

Reply:
fixed